packages/lit-html/src/directives/keyed.ts

```diff
@@ -0,0 +1,37 @@
+import {nothing} from '../lit-html.js';
+import {
+  directive,
+  Directive,
+  ChildPart,
+  DirectiveParameters,
+} from '../directive.js';
+import {setCommittedValue} from '../directive-helpers.js';
+
+class Keyed extends Directive {
+  key: unknown = nothing;
+
+  render(k: unknown, v: unknown) {
+    this.key = k;
+    return v;
+  }
+
+  override update(part: ChildPart, [k, v]: DirectiveParameters<this>) {
+    if (k !== this.key) {
+      // Clear the part before returning a value. This forces a re-render.
+      setCommittedValue(part);
+      this.key = k;
+    }
+    return v;
+  }
+}
+
+/**
+ * Associates a renderable value with a unique key. When the key changes, the
+ * part is cleared forcing a new render, even if the value - such as a template
```

Comment:
```suggestion
 * previous DOM is removed and disposed before rendering the next value, even if the value - such as a template
```

I think this clarifies the exact behavior better?